wasp/apps/alarm.py

```diff
@@ -0,0 +1,155 @@
+            if self.ringing:
+                self.ringing = False
+
+    def tick(self, ticks):
+        """Notify the application that its periodic tick is due."""
+        wasp.watch.vibrator.pulse(duty=50, ms=500)
+        wasp.system.keep_awake()
+
+    def touch(self, event):
+        """Notify the application of a touchscreen touch event."""
+        draw = wasp.watch.drawable
+        if event[1] in range(90, 150) and event[2] in range(180,240):
+            self.active = not self.active
+
+        elif event[1] in range(30,90):
+            if event[2] in range(40,100):
+                self.hours += 1
+                if self.hours > 23:
+                    self.hours = 0
+
+            elif event[2] in range(120,180):
+                self.hours -= 1
+                if self.hours < 0:
+                    self.hours = 23
+
+        elif event[1] in range(150,210):
+            if event[2] in range(40,100):
+                self.minutes += 1
+                if self.minutes > 59:
+                    self.minutes = 0
+
+            elif event[2] in range(120,180):
+                self.minutes -= 1
+                if self.minutes < 0:
+                    self.minutes = 59
+
+        self._update()
+
+    def _draw(self):
+        """Draw the display from scratch."""
+        draw = wasp.watch.drawable
+        if not self.ringing:
+            draw.fill()
+            draw.string(self.NAME, 0, 6, width=240)
+
+            draw.fill(0xffff, 120, 112, 2, 2)
+            draw.fill(0xffff, 120, 106, 2, 2)
+
+            for posx in [40,160]:
+                draw.string("+", posx, 60, width=40)
+                draw.string("-", posx, 140, width=40)
+
+            draw.fill(0xffff, 100, 190, 40, 40)
+
+            self._update()
+        else:
+            draw.fill()
+            draw.string("Alarm", 0, 150, width=240)
+            draw.blit(icon, 73, 50)
+            
+    def _update(self):
+        """Update the dynamic parts of the application display."""
+        draw = wasp.watch.drawable
+        if self.active:
+            draw.fill(0x001f, 102, 192, 36, 36)
+        else:
+            draw.fill(0x0000, 102, 192, 36, 36)
+        
+        if self.hours < 10:
+            draw.string("0"+str(self.hours), 10, 100, width=100)
+        else:
+            draw.string(str(self.hours), 10, 100, width=100)
+
+        if self.minutes < 10:
+            draw.string("0"+str(self.minutes), 130, 100, width=100)
+        else:
+            draw.string(str(self.minutes), 130, 100, width=100)
+
+    def _alert(self):
+        self.ringing = True
+        wasp.system.wake()
+        wasp.system.switch(self)
+
+    def _set_current_alarm(self):
+        now = wasp.watch.rtc.get_localtime()
+        yyyy = now[0]
+        mm = now[1]
+        dd = now[2]
+        if self.hours < now[3] or (self.hours == now[3] and self.minutes <= now[4]):
+            dd += 1
+        self.current_alarm = (yyyy, mm, dd, self.hours, self.minutes, 0, 0, 0, 0)
```

Comment:
It would be better to store the output of time.mktime in self.current_alarm (it will use less RAM when the app is in the background).